pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -0,0 +1,282 @@
+				return validation.ValidateInfrastructure(new)
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Infrastructure)
+				old := o.(*extensionsv1alpha1.Infrastructure)
+				return validation.ValidateInfrastructureUpdate(new, old)
+			}})
+
+	h.addResource(metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "networks"},
+		artifact{
+			newEntity: func() client.Object { return new(extensionsv1alpha1.Network) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Network)
+				return validation.ValidateNetwork(new)
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Network)
+				old := o.(*extensionsv1alpha1.Network)
+				return validation.ValidateNetworkUpdate(new, old)
+			}})
+
+	h.addResource(metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "operatingsystemconfigs"},
+		artifact{
+			newEntity: func() client.Object { return new(extensionsv1alpha1.OperatingSystemConfig) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.OperatingSystemConfig)
+				return validation.ValidateOperatingSystemConfig(new)
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.OperatingSystemConfig)
+				old := o.(*extensionsv1alpha1.OperatingSystemConfig)
+				return validation.ValidateOperatingSystemConfigUpdate(new, old)
+			}})
+
+	h.addResource(metav1.GroupVersionResource{Group: extensionsv1alpha1.SchemeGroupVersion.Group, Version: extensionsv1alpha1.SchemeGroupVersion.Version, Resource: "workers"},
+		artifact{
+			newEntity: func() client.Object { return new(extensionsv1alpha1.Worker) },
+			validateCreateResource: func(n, _ client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Worker)
+				return validation.ValidateWorker(new)
+			},
+			validateUpdateResource: func(n, o client.Object) field.ErrorList {
+				new := n.(*extensionsv1alpha1.Worker)
+				old := o.(*extensionsv1alpha1.Worker)
+				return validation.ValidateWorkerUpdate(new, old)
+			}})
+
+	return &h
+}
+
+type handler struct {
+	decoder   *admission.Decoder
+	logger    logr.Logger
+	artifacts map[metav1.GroupVersionResource]artifact
+}
+
+func (h *handler) addResource(gvr metav1.GroupVersionResource, art artifact) {
+	h.artifacts[gvr] = art
+}
+
+var _ admission.Handler = &handler{}
+
+func (h *handler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
+
+func (h *handler) Handle(ctx context.Context, ar admission.Request) admission.Response {
+	h.logger.Info(fmt.Sprintf("validating resource of type %s for operation %s", ar.Resource, ar.Operation))
```

Comment:
Nit: The more idiomatic way is to benefit from the key/value pairs that the logger accept as context:

```suggestion
	h.logger.Info("Validating resource", "type", ar.Resource, "operation", ar.Operation)
```